lib/libzpool/util.c

```diff
@@ -123,12 +123,15 @@ show_vdev_stats(const char *desc, const char *ctype, nvlist_t *nv, int indent)
 
 	for (c = 0; c < children; c++) {
 		nvlist_t *cnv = child[c];
-		char *cname, *tname;
+		char *cname = NULL;
+		char *tname = NULL;
 		uint64_t np;
 		if (nvlist_lookup_string(cnv, ZPOOL_CONFIG_PATH, &cname) &&
 		    nvlist_lookup_string(cnv, ZPOOL_CONFIG_TYPE, &cname))
 			cname = "<unknown>";
 		tname = calloc(1, strlen(cname) + 2);
+		if (tname == NULL || cname == NULL)
+			continue;
 		(void) strcpy(tname, cname);
 		if (nvlist_lookup_uint64(cnv, ZPOOL_CONFIG_NPARITY, &np) == 0)
 			tname[strlen(tname)] = '0' + np;
```

Comment:
A simpler way to handle this would be to reworked the function to use `umem_zalloc()` like this:

```diff
diff --git a/lib/libzpool/util.c b/lib/libzpool/util.c
index 7a0748c..11a84c5 100644
--- a/lib/libzpool/util.c
+++ b/lib/libzpool/util.c
@@ -123,13 +123,14 @@ show_vdev_stats(const char *desc, const char *ctype, nvlis
 
        for (c = 0; c < children; c++) {
                nvlist_t *cnv = child[c];
-               char *cname, *tname;
+               char *cname = NULL, *tname;
                uint64_t np;
+
                if (nvlist_lookup_string(cnv, ZPOOL_CONFIG_PATH, &cname) &&
                    nvlist_lookup_string(cnv, ZPOOL_CONFIG_TYPE, &cname))
                        cname = "<unknown>";
-               tname = calloc(1, strlen(cname) + 2);
-               (void) strcpy(tname, cname);
+               tname = umem_zalloc(strlen(cname) + 2, UMEM_NOFAIL);
+               strlcpy(tname, cname, strlen(cname) + 2);
                if (nvlist_lookup_uint64(cnv, ZPOOL_CONFIG_NPARITY, &np) == 0)
                        tname[strlen(tname)] = '0' + np;
```